src/logging.cpp

```diff
@@ -184,6 +184,93 @@ bool GetLogCategory(BCLog::LogFlags& flag, const std::string& str)
     return false;
 }
 
+std::string LogLevelToStr(const BCLog::Level& level)
+{
+    switch (level) {
+    case BCLog::Level::None:
+        return "none";
+    case BCLog::Level::Debug:
+        return "debug";
+    case BCLog::Level::Info:
+        return "info";
+    case BCLog::Level::Warning:
+        return "warning";
+    case BCLog::Level::Error:
+        return "error";
+    default:
+        return "none";
+    }
+
+}
+
+std::string LogCategoryToStr(const BCLog::LogFlags& category)
```

Comment:
It would be good to consistently pass both LogFlags and Level in this pull either by (non-const) value or by reference to const, but not both.

Reply:
I wonder at what point it makes sense to have an array of strings and use a `size_t` cast of LogFlags as an index.

Reply:
Thanks for the reference! Updated this pull to pass both LogFlags and Level by value.

Reply:
> I wonder at what point it makes sense to have an array of category strings and use a size_t cast of LogFlags as an index

@jonatack We chose switch-case as it provides us with a compile-time warning. Additionally, IMO for large enums, switch-cases are easier to review as compared to checking the index of the elements against the enum value. 

However, I do agree that this function could easily become bloated but I am not sure what is the best solution for it.

Reply:
Yes, I was musing more about performance as this code can be called frequently and an array lookup is constant time; maybe you can use the existing `CLogCategoryDesc`.

Reply:
Do you have any suggestions on how we should map the enums to the index on the array? Currently, the majority enums values of `LogFlags` are bit flags (ie `1<<i`) and the enum values of `LogFlags::NONE` and `LogFlags::ALL` are `0` and all bits sets respectively.